src/EluvioLive.js

```diff
@@ -2928,15 +2928,81 @@ class EluvioLive {
   /**
    * Generate tickets (NTP) as a tenant.
    *
+   * Tickets may be bound to emails - in this case, the list of emails has to be supplied
+   * as an argument, and the 'quantity' is no longer allowed.
+   * Optionally creates embed URLs for each ticket code generated, based on a template URL
+   * that contains all required embed URL parameters except for the ticket and optionally email.
+   *
    * @namedParams
    * @param {string} tenant - The Tenant ID
    * @param {string} otp - The OTP ID
+   * @param {integer} otpClass - The OTP class (4 or 5) (default 5)
    * @param {integer} quantity - Number of tickets to generate
+   * @param {string} emails - File containing one email per line
+   * @param {strnig} embedUrlBase - Base embed URL, used to generate embed URLs for each ticket
    * @return {Promise<Object>} - Tickets API Response Object
    */
-  async TenantTicketsGenerate({ tenant, otp, host, quantity = 1 }) {
+  async TenantTicketsGenerate({ tenant, otp, otpClass = 5, host, quantity, emails, embedUrlBase }) {
     let now = Date.now();
 
+    // Make embed URL from base URL by adding code and optional email
+    function makeEmbedUrl({code, email, embedUrlBase}) {
+      // Make embed URL from base URL
+      let u = new URL(embedUrlBase);
+      let code64 = Buffer.from(code).toString("base64");
+      u.searchParams.append("tk", code64);
+      if (email) {
+        let email64 = Buffer.from(email).toString("base64");
+        u.searchParams.append("sbj", email64);
+      }
+      let embedUrl = u.href;
+      return embedUrl;
+    }
+
+    // Validation - either emails or quantity may be specified
+    if (emails && quantity) {
+      throw "Only one of parameters emails and quantity may be specified";
+    }
+    if (!emails && !quantity) {
+      quantity = 1;
+    }
+
+    if (otpClass && otpClass == 4) {
+
+      var emailList = [];
+      if (emails) {
+        const emailsBuf = fs.readFileSync(emails, "utf-8");
+        emailsBuf.split(/\r?\n/).forEach((line) => {
+          if (line) {
```

Comment:
could be could to add some email validation here.  where is this file coming from and how often are there trash data or invalid emails?

Reply:
regex validation or something more

function checkMail(email){
	var filter  = /^([a-zA-Z0-9_\.\-])+\@(([a-zA-Z0-9\-])+\.)+([a-zA-Z0-9]{2,4})+$/;
	if (filter.test(email)) {
		return true;
	}
	return false;
}

Reply:
could be could to add some email validation here. where is this file coming from and how often are there trash data or invalid emails?